source/adapters/opencl/command_buffer.cpp

```diff
@@ -74,43 +134,82 @@ UR_APIEXPORT ur_result_t UR_APICALL urCommandBufferAppendMemcpyUSMExp(
         *pSyncPointWaitList,
     [[maybe_unused]] ur_exp_command_buffer_sync_point_t *pSyncPoint) {
 
-  cl_adapter::die("Experimental Command-buffer feature is not "
+  cl_adapter::die("Experimental Command-buffer entry point is not "
                   "implemented for OpenCL adapter.");
   return UR_RESULT_ERROR_UNSUPPORTED_FEATURE;
 }
 
-UR_APIEXPORT ur_result_t UR_APICALL urCommandBufferAppendMembufferCopyExp(
+UR_APIEXPORT ur_result_t UR_APICALL urCommandBufferAppendUSMFillExp(
     [[maybe_unused]] ur_exp_command_buffer_handle_t hCommandBuffer,
-    [[maybe_unused]] ur_mem_handle_t hSrcMem,
-    [[maybe_unused]] ur_mem_handle_t hDstMem, [[maybe_unused]] size_t srcOffset,
-    [[maybe_unused]] size_t dstOffset, [[maybe_unused]] size_t size,
+    [[maybe_unused]] void *pMemory, [[maybe_unused]] const void *pPattern,
+    [[maybe_unused]] size_t patternSize, [[maybe_unused]] size_t size,
     [[maybe_unused]] uint32_t numSyncPointsInWaitList,
     [[maybe_unused]] const ur_exp_command_buffer_sync_point_t
         *pSyncPointWaitList,
     [[maybe_unused]] ur_exp_command_buffer_sync_point_t *pSyncPoint) {
-
-  cl_adapter::die("Experimental Command-buffer feature is not "
+  cl_adapter::die("Experimental Command-buffer entry point is not "
                   "implemented for OpenCL adapter.");
   return UR_RESULT_ERROR_UNSUPPORTED_FEATURE;
 }
 
+UR_APIEXPORT ur_result_t UR_APICALL urCommandBufferAppendMembufferCopyExp(
+    ur_exp_command_buffer_handle_t hCommandBuffer, ur_mem_handle_t hSrcMem,
+    ur_mem_handle_t hDstMem, size_t srcOffset, size_t dstOffset, size_t size,
+    uint32_t numSyncPointsInWaitList,
+    const ur_exp_command_buffer_sync_point_t *pSyncPointWaitList,
+    ur_exp_command_buffer_sync_point_t *pSyncPoint) {
+
+  cl_context CLContext = cl_adapter::cast<cl_context>(hCommandBuffer->hContext);
+  cl_ext::clCommandCopyBufferKHR_fn clCommandCopyBufferKHR = nullptr;
+  cl_int Res = cl_ext::getExtFuncFromContext<decltype(clCommandCopyBufferKHR)>(
+      CLContext, cl_ext::ExtFuncPtrCache->clCommandCopyBufferKHRCache,
+      cl_ext::CommandCopyBufferName, &clCommandCopyBufferKHR);
+
+  if (!clCommandCopyBufferKHR || Res != CL_SUCCESS)
+    return UR_RESULT_ERROR_INVALID_OPERATION;
+
+  CL_RETURN_ON_FAILURE(clCommandCopyBufferKHR(
+      hCommandBuffer->CLCommandBuffer,
+      cl_adapter::cast<cl_command_queue>(hCommandBuffer->hInternalQueue),
```

Comment:
From the spec wording from `clCommandCopyBufferKHR` for where we are passing the internal queue:

> command_queue Specifies the command-queue the command will be recorded to. Parameter is unused by this extension as only a single command-queue is supported and must be NULL.

Similar wording for other entry-points. We should replace this internal queue parameter passed on command-recording entry-points with NULL

Reply:
Fixed: `clCommandCopyBufferKHR`, `clCommandCopyBufferRectKHR` and `clCommandFillBufferKHR`